<p style="text-align: center; font-weight: bold; font-size: 18pt">Loading Data</p>

In [9]:
library(repr)
library(tidyverse)

In [14]:
download.file(url = "https://raw.githubusercontent.com/MatthewKyi/DSCI-100-004-Individual/refs/heads/main/players.csv", destfile = "players.csv")
players <- read_csv("players.csv")
head(players)

distinct(players, experience)
distinct(players, gender)

"Mean"
map_df(select(players, subscribe, played_hours, Age), mean, na.rm=TRUE)
"Min"
map_df(select(players, played_hours, Age), min, na.rm=TRUE)
"Max"
map_df(select(players, played_hours, Age), max, na.rm=TRUE)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

[1] "Mean"

subscribe played_hours Age     
1 0.7346939 5.845918     21.13918

[1] "Min"

played_hours Age
1 0            9

[1] "Max"

played_hours Age
1 223.1        58

In [15]:
download.file(url = "https://raw.githubusercontent.com/MatthewKyi/DSCI-100-004-Individual/refs/heads/main/sessions.csv", destfile = "sessions.csv")
sessions <- read_csv("sessions.csv")
head(sessions)

"Mean"
map_df(select(sessions, original_start_time, original_end_time), mean, na.rm=TRUE)
"Min"
map_df(select(sessions, original_start_time, original_end_time), min, na.rm=TRUE)
"Max"
map_df(select(sessions, original_start_time, original_end_time), max, na.rm=TRUE)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

[1] "Mean"

original_start_time original_end_time
1 1.719201e+12        1.719196e+12

[1] "Min"

original_start_time original_end_time
1 1.7124e+12          1.7124e+12

[1] "Max"

original_start_time original_end_time
1 1.72733e+12         1.72734e+12

<p style="text-align: center; font-weight: bold; font-size: 18pt">Data Explanation</p>

The data in this project relates to the PlaiCraft Minecraft server, and is meant to provide information on how people play video games. There are two relevant datasets. One of which includes information about each player in the Minecraft server, and the other information about all the player's gaming sessions on the server.
<br><br>
<b style="font-size: 14pt">players dataset</b>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- 7 columns/variables, 196 rows/observations<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Approximately 73.47% of players are subscribed<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Average played_hours = 5.84<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Minimum played_hours = 0<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Maximum played_hours = 223.10<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Average player age is 21.13<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Minimum player age is 9<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Maximum player age is 58<br></p>
<br><br>
<b style="font-size: 14pt">sessions dataset</b>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- 5 columns/variables, 1535 rows/observations<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Average original_start_time = 1.719201e+12ms<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Earliest original_start_time = 1.7124e+12ms<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Latest original_start_time = 1.72733e+12ms<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Average original_end_time = 1.719196e+12ms<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Earliest original_end_time = 1.7124e+12ms<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Latest original_end_time = 1.72734e+12ms<br></p>

<p style="text-align: center; font-weight: bold; font-size: 18pt">Variables Explanation</p>

The dataset of *players* has 7 columns/variables: <br>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- experience (Character): Pro, Veteran, Regular, Beginner, or Amateur <br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- subscribe (Logical): Is the player subscribed -- TRUE or FALSE <br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- hashedEmail (Character): SHA256 encrypted email address of player (196 unique hashed emails)<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- played_hours (Double): how many hours has the player been online <br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- name (Character): player's name<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- gender (Character): Male, Female, Non-binary, Agender, Two-Spirited, Other, Prefer not to say<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- Age (Double): player's age<br></p>
<br><br>
The dataset of *sessions* has 5 columns/variables:<br>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- hashedEmail (Character): SHA256 encrypted email address of player (125 unique hashed emails)<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- start_time (Character): session start time in dd/mm/yyyy hh:mm format<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- end_time (Character): session end time in dd/mm/yyyy hh:mm format<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- original_start_time (Double): time of session start in UNIX Time (Time since Jan 1st 1970) in milliseconds<br></p>
<p>&nbsp;&nbsp;&nbsp;&nbsp;- original_end_time (Double): elapsed of session end in UNIX Time (Time since Jan 1st 1970) in milliseconds</p>

<p style="text-align: center; font-weight: bold; font-size: 18pt">Potential Issues</p>

<p>- The hashedEmail variable is somewhat difficult to work with as opposed to something like username to keep track of which player is which <br></p>
<p>- The original_start_time and original_end_time variables are only measured to 6 signifcant figures, which is not enough precision to show any difference between the two. <br></p>
<p>- The start_time and end_time variables are of type character, and will take some effort to find the difference in time between the two if that is necessary.<br></p>
<p>- The data may not consider Away from Keyboard (AFK) time of players in the session dataframe<br></p>
<p>- As opposed to something like a years played variable, the experience variable is somewhat subjective for measuring player skill <br></p>


<p style="text-align: center; font-weight: bold; font-size: 18pt">Question</p>

<h4>Broad question</h4>
We are interested in demand forecasting, namely, what time windows are most likely to have large number of simultaneous players. This is because we need to ensure that the number of licenses on hand is sufficiently large to accommodate all parallel players with high probability. 

<h4>Specific question</h4>
Can the current date/time predict the number of active players in the sessions dataset?
<br>
<p>The sessions dataset contains session time information. If there are many sessions at particular times, we know that those times are more likely to have a high volume of players, and vice. versa. It may help to use the start_time and end_time variable, and convert them into numeric UNIX time such as the original_start_time and original_end_time but with more percision.</p>